app/build.gradle.kts

```
@@ -0,0 +1,304 @@
+/*
+ * SPDX-FileCopyrightText: 2025 NewPipe e.V. <https://newpipe-ev.de>
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+plugins {
+    alias(libs.plugins.android.application)
+    alias(libs.plugins.jetbrains.kotlin.android)
+    alias(libs.plugins.jetbrains.kotlin.kapt)
+    alias(libs.plugins.jetbrains.kotlin.parcelize)
+    alias(libs.plugins.sonarqube)
+    checkstyle
+}
+
+apply(from = "check-dependencies.gradle.kts")
+
+val gitWorkingBranch = providers.exec {
+    commandLine("git", "rev-parse", "--abbrev-ref", "HEAD")
+}.standardOutput.asText.map { it.trim() }
+
+android {
+    compileSdk = 36
+    namespace = "org.schabi.newpipe"
+
+    defaultConfig {
+        applicationId = "org.schabi.newpipe"
+        resValue("string", "app_name", "NewPipe")
+        minSdk = 21
+        targetSdk = 35
+
+        versionCode = System.getProperty("versionCodeOverride")?.toInt() ?: 1005
+
+        versionName = "0.28.0"
+        System.getProperty("versionNameSuffix")?.let { versionNameSuffix = it }
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        javaCompileOptions {
+            annotationProcessorOptions {
+                arguments["room.schemaLocation"] = "$projectDir/schemas"
+            }
+        }
+    }
+
+    buildTypes {
+        debug {
+            isDebuggable = true
+
+            // suffix the app id and the app name with git branch name
+            val defaultBranches = listOf("master", "dev")
+            val workingBranch = gitWorkingBranch.getOrElse("")
+            val normalizedWorkingBranch = workingBranch
+                .replaceFirst("^[^A-Za-z]+".toRegex(), "")
+                .replace("[^0-9A-Za-z]+".toRegex(), "")
+
+            if (normalizedWorkingBranch.isEmpty() || workingBranch in defaultBranches) {
+                // default values when branch name could not be determined or is master or dev
+                applicationIdSuffix = ".debug"
+                resValue("string", "app_name", "NewPipe Debug")
+            } else {
+                applicationIdSuffix = ".debug.$normalizedWorkingBranch"
+                resValue("string", "app_name", "NewPipe $workingBranch")
+                setProperty("archivesBaseName", "NewPipe_$normalizedWorkingBranch")
+            }
+        }
+
+        release {
+            System.getProperty("packageSuffix")?.let { suffix ->
+                applicationIdSuffix = suffix
+                resValue("string", "app_name", "NewPipe $suffix")
+                setProperty("archivesBaseName", "NewPipe_" + System.getProperty("packageSuffix"))
+            }
```

Here it is important to keep archivesBaseName because we keep it in 

Also:
```suggestion
            System.getProperty("packageSuffix")?.let { suffix ->
                applicationIdSuffix = suffix
                resValue("string", "app_name", "NewPipe $suffix")
                setProperty("archivesBaseName", "NewPipe_$suffix")
            }
```